parcels/particlesets/particlesetsoa.py

```diff
@@ -0,0 +1,528 @@
+
+        if self.repeatdt:
+            if len(time) > 0 and time[0] is None:
+                self.repeat_starttime = time[0]
+            else:
+                if self._collection.data['time'][0] and not np.allclose(self._collection.data['time'], self._collection.data['time'][0]):
+                    raise ValueError('All Particle.time should be the same when repeatdt is not None')
+                self.repeat_starttime = self._collection.data['time'][0]
+            self.repeatlon = self._collection.data['lon']
+            self.repeatlat = self._collection.data['lat']
+            self.repeatdepth = self._collection.data['depth']
+            for kwvar in kwargs:
+                self.repeatkwargs[kwvar] = self._collection.data[kwvar]
+
+        if self.repeatdt:
+            if MPI and self._collection.pu_indicators is not None:
+                mpi_comm = MPI.COMM_WORLD
+                mpi_rank = mpi_comm.Get_rank()
+                self.repeatpid = pid_orig[self._collection.pu_indicators == mpi_rank]
+
+        self.kernel = None
+
+    def _set_particle_vector(self, name, value):
+        """Set attributes of all particles to new values.
+
+        :param name: Name of the attribute (str).
+        :param value: New value to set the attribute of the particles to.
+        """
+        self.collection._data[name][:] = value
+
+    def _impute_release_times(self, default):
+        """Set attribute 'time' to default if encountering NaN values.
+
+        :param default: Default release time.
+        :return: Minimum and maximum release times.
+        """
+        if np.any(np.isnan(self._collection.data['time'])):
+            self._collection.data['time'][np.isnan(self._collection.data['time'])] = default
+        return np.min(self._collection.data['time']), np.max(self._collection.data['time'])
+
+    def data_indices(self, variable_name, compare_values, invert=False):
+        """Get the indices of all particles where the value of
+        `variable_name` equals (one of) `compare_values`.
+
+        :param variable_name: Name of the variable to check.
+        :param compare_values: Value or list of values to compare to.
+        :param invert: Whether to invert the selection. I.e., when True,
+                       return all indices that do not equal (one of)
+                       `compare_values`.
+        :return: Numpy array of indices that satisfy the test.
+        """
+        compare_values = np.array([compare_values, ]) if type(compare_values) not in [list, dict, np.ndarray] else compare_values
+        return np.where(np.isin(self._collection.data[variable_name], compare_values, invert=invert))[0]
+
+    def indexed_subset(self, indices):
+        return ParticleCollectionIteratorSOA(self._collection,
+                                             subset=indices)
```

Comment:
I believe this function is not currently used internally (correct me if I am wrong). Do we want to leave this in as a way (for the user) to interact with a subset of the particles?